website/docs/advanced-concepts/version-control-with-git/README.mdx

```diff
@@ -0,0 +1,102 @@
+---
+sidebar_position: 5
+---
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Version Control With Git
+
+Version Control with Git allows you to version control your apps by connecting them with a Git hosting provider (Github, GitLab, Bitbucket), making it easier to track changes, create rollbacks or collaborate using git branches.
+
+:::info
+In the community edition, You can connect up to **three** private repositories in a workspace. If you wish to connect more, you can upgrade to the [**business** edition](https://www.appsmith.com/pricing). However, you can connect **unlimited** public repositories.
+:::
+
+<VideoEmbed host="youtube" videoId="emdZaG-tzgw" title="Version Control with Git " caption="Version Control with Git "/>
+
+Version Control works with any Git hosting service that supports **SSH protocol** and **deploy keys**. HTTPS Git connections are currently not supported on Appsmith.
+
+## Connect to a Git repository
+
+The first step to use Version control in Appsmith is to connect your remote repository with the app. Please follow the steps given below:
+
+### Create a new repository
+
+Create a new repository on your git service and copy the `SSH` URL. The repository should be empty except `README.md`, `.gitignore`, and `LICENSE` files (These can be auto-created when you create a new repository.)
+
+### Generate a deploy key
+
+1. Open the app you want to connect to git and click on the `Connect GIT` button on the bottom bar.
+2. In the Git connection window, paste your repository’s `SSH` URL and click on `Generate key` to generate an `SSH` key.
+
+
+
+ <VideoEmbed host="youtube" videoId="oFfdIwhSPL8" title="Generating a Deploy key" caption="Generating a Deploy key"/>
+
+Appsmith supports two types of keys -  **ECDSA 256** and **RSA 4096**. Please pick the one that's compatible with your Git hosting service. 
+
+### Adding the deploy key in the Repository
+
+Copy the generated deploy key and add it to your repository. Please note that you have to **enable write access** for pushing the changes to the remote repository from Appsmith.
+
+<Tabs queryString="current-edition">
+<TabItem label="GitHub" value="github">
+
+* Open the settings of the repository and go to deploy keys.
+* Click **Add deploy key**. Add a title for the key and paste the generated key into the key section.
+* Check the **Allow write access** checkbox and add the key.
+
+  <VideoEmbed host="youtube" videoId="4xMHO4G2hEA" title="Adding the deploy key in the Repository | Github" caption="Adding the deploy key in the Repository | Github"/>
+
+</TabItem>
+<TabItem label="Gitlab" value="gitlab">
+
```

Comment:
Can we ensure the UI elements below in both GitLab and Bitbucket are styles as per Google style guide?